### ISSUE TITLE
Can't resolve absolute import of module stubs

### ISSUE THREAD

**Issue by itsadok** [microsoft/pyright#2691]
**Describe the bug**
Absolute imports of a module that is only available as a stub gets reported as "could not be resolved", whereas a relative import succeeds.
This is especially critical when the stubbed module is a top-level module, which cannot be relatively imported.

**To Reproduce**

```bash
$ mkdir /tmp/project
$ cd /tmp/project
$ mkdir pkg
$ touch pkg/module.pyi
$ echo "import pkg.module" > test.py
$ pyright test.py
...
  /tmp/project/test.py:1:8 - warning: Import "pkg.module" could not be resolved from source (reportMissingModuleSource)
0 errors, 1 warning, 0 infos
Completed in 0.837sec
$ echo "from pkg import module" > test.py
$ pyright test.py
...
0 errors, 0 warnings, 0 infos
Completed in 0.75sec
$ python --version
Python 3.9.5
$ pyright --version
pyright 1.1.194
```

**Expected behavior**
At the very least, the behavior should be consistent between the different ways to import the same module.

If I rename `module.pyi` to `module.py`, this warning goes away. 

I'm using venv and running on ubuntu 21.04, if that makes any difference.

**Comment by erictraut** [comment]
The `reportMissingModuleSource` diagnostic check is telling you that a corresponding runtime file (e.g. ".py" or ".so") couldn't be found, so if you run the code, it will crash at runtime.

If you don't care about this diagnostic check, you can create a pyrightconfig.json file and specify 
```json
{
    "reportMissingModuleSource": false
}
```

This check is applied only for some import forms — those that are typically used for importing packages that are installed in your Python environment. I agree that this appears to be inconsistent. We should probably never emit this error for files that are imported from your project _or_ emit this error for all import statement forms.

I'm going to transfer this bug to the pylance-release repo because this diagnostic check was added by the pylance team and is intended primarily for pylance users.
